concordium-sdk/src/main/java/com/concordium/sdk/crypto/wallet/web3Id/Statement/UnqualifiedRequestStatement.java

```diff
@@ -0,0 +1,47 @@
+package com.concordium.sdk.crypto.wallet.web3Id.Statement;
+
+import org.apache.commons.codec.binary.Hex;
+
+import com.concordium.sdk.crypto.ed25519.ED25519PublicKey;
+import com.concordium.sdk.crypto.wallet.Network;
+import com.concordium.sdk.transactions.CredentialRegistrationId;
+import com.concordium.sdk.types.ContractAddress;
+
+import lombok.Getter;
+import lombok.experimental.SuperBuilder;
+import lombok.extern.jackson.Jacksonized;
+
+@Getter
+@SuperBuilder
+@Jacksonized
+public class UnqualifiedRequestStatement extends RequestStatement {
+    private IdQualifier idQualifier;
+
+    public QualifiedRequestStatement qualify(CredentialRegistrationId credId, Network network) {
+        if (!idQualifier.getType().equals(IdQualifier.QualifierType.Credential)) {
+            throw new IllegalArgumentException("Only an account statement may only be qualified using a credentialId");
+        }
+        String did = "did:ccd:" + network.getValue().toLowerCase() + ":cred:" + credId.getEncoded();
+        return QualifiedRequestStatement.builder().id(did).statement(getStatement()).build();
+    }
+
+    public QualifiedRequestStatement qualify(ContractAddress contract, ED25519PublicKey publicKey, Network network) {
+        if (!idQualifier.getType().equals(IdQualifier.QualifierType.SmartContract)) {
+            throw new IllegalArgumentException(
+                    "Only an web3Id statement may only be qualified using a Contract Address");
+        }
+        if (!((Web3IdIssuerQualifier) idQualifier).getIssuers().contains(contract)) {
```

Comment:
If we check this here, then perhaps we should also do the equivalent check when qualifying an account credential statement.

Reply:
Yes, but then we would just need to add more info, I just added it here because we can check it with only the info needed to qualify the statement.